File: Turnierplaner-Frontend/src/app/Objects/participant.ts
export class Participant {
private participantId;
private participantFirstName;
private participantLastName;
private participantResidence;
private participantDateOfBirth;

  constructor( participantFirstName: string, participantLastName: string, participantDateOfBirth: string, participantResidence: string)
  {
    this.participantDateOfBirth = participantDateOfBirth;
    this.participantFirstName = participantFirstName;
    this.participantLastName = participantLastName;
    this.participantResidence = participantResidence;

  }

  setParticipantId(id : number){
    this.participantId = id;
  }


}

File: Turnierplaner-Backend/src/main/java/ch/zhaw/psit4/turnier/model/Participant.java
package ch.zhaw.psit4.turnier.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

import javax.persistence.*;
import java.sql.Date;
import java.util.HashSet;
import java.util.Set;


@Data
@Entity
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Participant {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer participantId;

    private String participantFirstName;

    private String participantLastName;

    private String participantResidence;

    private String participantZipCode;

    private Date participantDateOfBirth;

    @JsonIgnore
    @ToString.Exclude
    @EqualsAndHashCode.Exclude
    @OneToMany(cascade = CascadeType.ALL, fetch = FetchType.LAZY, mappedBy = "participant")
    private Set<TournamentParticipant> tournamentParticipant = new HashSet<>();


}
